2020q2/multiarch-ci.md

```diff
@@ -0,0 +1,34 @@
+## Continuous Integration on !x86 ##
+
+Contact: Edward Tomasz Napierala, <trasz@FreeBSD.org>
+
+For quite a while the [FreeBSD CI infrastructure](https://ci.FreeBSD.org)
+has been running FreeBSD builds and regression tests, making it easy to
+spot regressions.  While CI was building images for all architectures,
+the regression tests were only run on amd64 and i386, which means they
+couldn't detect architecture-specific runtime breakage.  The goal of this
+project is to run them for remaining architectures - ARM, ARM64, MIPS64,
+POWER, and RISC-V.
+
+The tests are being run using common, mostly machine-independent scripts.
+Those required some changes to make it possible to use QEMU in addition
+to Bhyve, the hypervisor used for x86 tests.  The `sysutils/u-boot-qemu-arm`
+and `sysutils/u-boot-qemu-arm64` ports were added to the Ports Collection.
+Finally, each of the architectures required some tweaks, to account for
+different configuration requirements - for example the MIPS64 kernel
+doesn't support VirtIO disks, or even AHCI, whereas on ARM64 the U-Boot
+gets confused with more than one VirtIO disk.
```

Comment:
At the risk of creating more work .. we should have in mind QEMU-userlevel at some point as well. But not today.

Reply:
Tbh I'm not sure about that - the main thing we want to test is the kernel, not just userspace.